lib/rucio/core/rule.py

```diff
@@ -1849,27 +1849,29 @@ def update_rules_for_bad_replica(scope, name, rse_id, nowait=False, session=None
 
 
 @transactional_session
-def generate_rule_notifications(rule, session=None):
+def generate_rule_notifications(rule, replicating_locks_before=None, session=None):
     """
-    Generate (If necessary) a callback for a rule (DATASETLOCK_OK, RULE_OK)
+    Generate (If necessary) a callback for a rule (DATASETLOCK_OK, RULE_OK, DATASETLOCK_PROGRESS)
 
-    :param rule:     The rule object.
-    :param session:  The Database session
+    :param rule:                       The rule object.
+    :param replicating_locks_before:   Amount of replicating locks before the current state change.
+    :param session:                    The Database session
     """
 
     session.flush()
+    total_locks = rule.locks_replicating_cnt + rule.locks_ok_cnt
```

Comment:
Yes. The conveyor does not trigger callbacks for rules who are in STUCK mode and the judge-repairer only triggers the callback once the rules state has changed to REPLICATING again. Thus this will never be called for STUCK rules.